lib/simple_crowd/client.rb

```diff
@@ -151,6 +151,8 @@ def search_users_by_email email
     # For example:
     #   client.search_users(:email => 'foo', :display_name => 'bar')
     def search_users criteria, limit=0, start=0
+      # yolk#201 : if email too short return [] in order to avoid high memory consumption as crowd will do partial match on email
+      return [] if criteria && criteria.stringify_keys.keys == ['email'] && criteria.stringify_keys['email'].length < 4
```

Comment:
@paloc I moved the check for the email length into the search_users and simply don't trigger the search and return an empty array in order to not be disruptive for the clients